dev/tests/integration/testsuite/Magento/Customer/Model/AccountManagement/CreateAccountTest.php

```diff
@@ -287,6 +303,56 @@ public function testCreateNewCustomerWithPasswordHash(): void
         );
     }
 
+    /**
+     * @dataProvider testCreateNewCustomerConfirmationKeyProvider
+     * @param string $expected
+     * @param string $is_confirmation_required
+     * @return void
+     */
+    public function testCreateNewCustomerConfirmation($expected, $is_confirmation_required): void
+    {
+        $website_id = 1;
+        $this->configWriter->save(
+            AccountConfirmation::XML_PATH_IS_CONFIRM,
+            $is_confirmation_required,
+            ScopeInterface::SCOPE_WEBSITES,
+            $website_id
+        );
+        $this->scopeConfig->clean();
```

Comment:
Writing to config inside test itself isn't recommended way, as it have to be cleaned.
Could you use build in feature (that cleans all changed configs automatically) for defining config values and split it to two test cases?
 https://devdocs.magento.com/guides/v2.4/test/integration/annotations/magento-config-fixture.html

Reply:
Hello @ihor-sviziev ,

I know of the annotation, however, the annotation does not work for website-scope config and as you can see here the function the determines whether confirmation is required or not uses website scope:

https://github.com/magento/magento2/blob/4b75b15706170d487b2615dffa0215af825e614a/app/code/Magento/Customer/Model/AccountConfirmation.php#L54-L65

I've tried the annotation for current_store as well as default, both don't work for website scope. Do you have any other suggestions?

Reply:
I think value should be set on global scope, similar to this (prefixed with `default/`:
https://github.com/magento/magento2/blob/40a787632f36ca67f8fff8c451e1e8d1bdbaf9f6/dev/tests/integration/testsuite/Magento/Catalog/Block/Product/View/MultiStoreCurrencyTest.php#L34

Reply:
I have already tried that and it works for SCOPE_TYPE_DEFAULT, however not for SCOPE_WEBSITES which isConfirmationRequired function uses.

Reply:
Interesting... usually website scope inherits all config values from default scope. Maybe in case of integration tests it works a bit incorrectly.

Ok, in this case let's leave it as is for now, but please fix permissions, variable names issues + define types for arguments in method